### DISCUSSION TITLE
Does not refresh the the entities

### DISCUSSION THREAD

**Discussion by geko100** [Thomas55555/husqvarna_automower#420] [Q&A]
Hello,

at first thanks for the great integration.
I got the problem, that I don't get new values. Only if I reload the integration it updates it.
What am I doing wrong?
Thanks!

**Comment by Thomas55555** [comment]
Maybe the websocket is not working. Can you download the diagnostics of the
integration and check if there is something like `amc` in the token scope?

geko100 ***@***.***> schrieb am Di., 11. Apr. 2023, 18:37:

**Comment by geko100** [comment]
Thats a part of the Log and I censored some sensitive parts (as I think) with ??????????????? 
I didn't find anything with 'amc':


"config_entry": {
      "entry_id": "??????????????????????",
      "version": 2,
      "domain": "husqvarna_automower",
      "title": "husqvarna_automower",
      "data": {
        "auth_implementation": "husqvarna_automower_??????????????????????",
        "token": {
          "access_token": "**REDACTED**",
          "scope": "iam:read",
          "expires_in": 86399,
          "refresh_token": "**REDACTED**",
          "provider": "husqvarna",
          "user_id": "??????????????????????",
          "token_type": "Bearer",
          "expires_at": 1681276329.826706
        }
      },
      "options": {
        "enable_camera": true,
        "gps_top_left": [
          ??????????????????????,
          ??????????????????????
        ],
        "gps_bottom_right": [
          ??????????????????????,
          ??????????????????????
        ],
        "mower_img_path": "/config/custom_components/husqvarna_automower/resources/mower.png",
        "map_img_path": "/config/custom_components/husqvarna_automower/resources/garten.png"
      },
      "pref_disable_new_entities": false,
      "pref_disable_polling": false,
      "source": "user",
      "unique_id": "husqvarna_automower",
      "disabled_by": null
    },

**Comment by Thomas55555** [comment]
You need the right scope for the token. Check the description here:
https://developer.husqvarnagroup.cloud/apis/automower-connect-api#websocket

geko100 ***@***.***> schrieb am Di., 11. Apr. 2023, 19:41:

**Reply by geko100** [reply]
As I wrote in my last reply, it works better, but I get only a few updates. Any ideas?

**Reply by Thomas55555** [reply]
Can you share your debug logs?

**Reply by geko100** [reply]
Sorry, but how to I get those Debug log?

**Comment by geko100** [comment]
Thanks. I deleted everything and did it again and now I got this at scope:

 "scope": "iam:read amc:api",

Tomorrow I will see if it works.

**Comment by geko100** [comment]
Now it works better. I see the right status and also the mower on the map but only with very less updates, only every hour I think or something.

**Comment by Thomas55555** [comment]
https://github.com/Thomas55555/husqvarna_automower#debugging

geko100 ***@***.***> schrieb am Di., 18. Apr. 2023, 15:43:

**Reply by geko100** [reply]
Is it possible to send it as email to you or something? I don't know if there are also sensitive data in the log and in this log is more then only your debug log.

**Reply by Thomas55555** [reply]
Yes. If you give me your email address, I can write you, so you have my address.
